Add mcxa156 lptmr support

NeilChen93 opened the pull request:
Add mcxa156 lptmr support

NeilChen93 committed:
boards: nxp: frdm_mcxa156: update doc index

Update board overview and hardware configuration

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

mmahadevan108 commented on boards/nxp/frdm_mcxa156/frdm_mcxa156_defconfig:
```diff
@@ -9,3 +9,4 @@ CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_GPIO=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=96000000
```
I would agree that this change so at the least be in a separate commit with details on why this change is needed.

NeilChen93 committed:
soc: mcxa156: update systick clock frequency to 96MHz

MCXA156 max frequeny is 96MHz

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
dts: arm/nxp: Add lptmr nodes to NXP MCXA156 dtsi file

Add lptmr nodes to NXP MCXA156 dtsi file

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
boards: nxp: frdm_mcxa156: Support lptmr for NXP frdm_mcxa156 board

Support lptmr for NXP frdm_mcxa156 board.
Test using tests/drivers/counter/counter_basic_api.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>
